packages/pigeon/lib/swift_generator.dart

```diff
@@ -576,11 +578,10 @@ class SwiftGenerator extends StructuredGenerator<SwiftOptions> {
     if (type.isNullable) {
       if (type.isClass) {
         indent.writeln('var $variableName: $fieldType? = nil');
-        indent
-            .write('if let ${variableName}List: [Any?] = nilOrValue($value) ');
+        indent.write(
+            'if let ${variableName}List: $fieldType = nilOrValue($value) ');
         indent.addScoped('{', '}', () {
-          indent.writeln(
-              '$variableName = $fieldType.fromList(${variableName}List)');
+          indent.writeln('$variableName = ${variableName}List as $fieldType');
```

Comment:
Shouldn't these no longer have `List` in the name?

(Also, is this still actually different than the `else` branch?)